apollo-backend/src/main/java/io/logz/apollo/controllers/StatusController.java

```diff
@@ -139,7 +152,11 @@ public KubernetesDeploymentStatus getOneSpecificStatusWithGroup(int envId, int s
         if (service != null) {
             List<Group> groups = groupDao.getGroupsPerServiceAndEnvironment(serviceId,envId);
             try {
-                groups.forEach(group -> kubernetesDeploymentStatuses.add(kubernetesHandler.getCurrentStatus(service, Optional.of(group.getName()))));
+                groups.forEach(group -> {
+                    KubernetesDeploymentStatus kubernetesDeploymentStatus = kubernetesHandler.getCurrentStatus(service, Optional.of(group.getName()));
+                    kubernetesDeploymentStatus = addCommitUrlToStatus(kubernetesDeploymentStatus, serviceId);
```

Comment:
This is a dangerous method - you are working by ref, and still have a return value.
No one will assume you are doing that.
If the method has a return value, it should not touch the original object (this is the preferred way).

Please change that (a couple of different places)